pkg/platformconfig/platformconfig.go

```diff
@@ -310,42 +310,49 @@ func (c *Config) GetDefaultWindowSizePresets() []string {
 // resources defined in the platform config, only if they are not already configured
 func (c *Config) EnrichContainerResources(ctx context.Context,
 	logger logger.Logger,
-	resources *v1.ResourceRequirements) {
+	resources *v1.ResourceRequirements,
+	isSideCar bool) {
```

Comment:
instead of giving `isSideCar` bool, id break this function into 2, reusing common functionality. that way, u wont get lost with the flow